From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 9 Sep 2026 13:16:24 +0530
Subject: [PATCH 087/109] Cygwin: restore the Windows DNS resolver
 backend on AArch64

(cherry picked from commit 767fa84bb8f46ed784e9e38fd437eceb57cd6a3e)
---
 winsup/cygwin/libc/minires-os-if.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/winsup/cygwin/libc/minires-os-if.c b/winsup/cygwin/libc/minires-os-if.c
index c1434b4b2..4e8e9cf21 100644
--- a/winsup/cygwin/libc/minires-os-if.c
+++ b/winsup/cygwin/libc/minires-os-if.c
@@ -12,15 +12,13 @@ details. */
 #define USE_SYS_TYPES_FD_SET
 #include <winsup.h>
 #include <ws2tcpip.h>
-
-#if defined(__CYGWIN__) && !defined(__aarch64__)
 #include <iphlpapi.h>
 #include <windns.h>
 #include "ntdll.h"
 #undef h_errno
 #include "minires.h"
 
-
+#ifdef __CYGWIN__
 /***********************************************************************
  *
  Windows interface code
@@ -504,8 +502,7 @@ void get_dns_info(res_state statp)
  Default interface code
 
 ***********************************************************************/
-struct __res_state;
-typedef struct __res_state *res_state;
+
 void get_dns_info(res_state statp)
 {
   return;
